drivers/modem/murata-1sc.c

```diff
@@ -0,0 +1,4499 @@
+#endif
+
+/**
+ * @brief Thread to process all messages received from the Modem.
+ */
+static void murata_1sc_rx(void)
+{
+	while (true) {
+		/* Wait for incoming data */
+		modem_iface_uart_rx_wait(&mctx.iface, K_FOREVER);
+
+		modem_cmd_handler_process(&mctx.cmd_handler, &mctx.iface);
+	}
+}
+
+/**
+ * @brief Convert string to long integer, but handle errors
+ */
+static int murata_1sc_atoi(const char *s, const int err_value, const char *desc, const char *func)
+{
+	int ret;
+	char *endptr;
+
+	ret = (int)strtol(s, &endptr, 10);
+	if (!endptr || *endptr != '\0') {
+		LOG_ERR("bad %s '%s' in %s", s, desc, func);
+		return err_value;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Convert ascii hex to uint8_t
+ */
+static inline uint8_t hex_char_to_int(char ch)
+{
+	uint8_t ret;
+
+	if (ch >= '0' && ch <= '9') {
+		ret = ch - '0';
+	} else if (ch > 'a' && ch < 'f') {
+		ret = 0xa + ch - 'a';
+	} else if (ch > 'A' && ch < 'F') {
+		ret = 0xa + ch - 'A';
+	} else {
+		ret = 0;
+	}
+	return ret;
+}
+
+/**
+ * @brief Derive a MAC address from the IMEI
+ */
+static inline uint8_t *murata_1sc_get_mac(const struct device *dev)
+{
+	struct murata_1sc_data *data = dev->data;
+
+	/* We will use the least significant 12 digits of the IMEI as
+	 * the MAC address. To explain further...
+	 * IMEI is always 15 digits long. The constant, MDM_IMEI_LENGTH
+	 * is 16 (bytes long) to hold the 15-digit IMEI plus terminator.
+	 * MAC address is always 6 bytes long (ie, 12 hex characters).
+	 * The loop below is calculating each of the MAC's 6 bytes
+	 */
+	for (int i = 0; i < 6; i++) {
+		int imei_idx = (MDM_IMEI_LENGTH - 1) - 12 + (i * 2);
+
+		data->mac_addr[i] = (hex_char_to_int(mdata.mdm_imei[imei_idx]) << 4) |
+				    (hex_char_to_int(mdata.mdm_imei[imei_idx + 1]));
+	}
+	return data->mac_addr;
+}
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+/**
+ * @brief Get the index from a specific socket pointer
+ *
+ * @param sock Socket pointer
+ * @return int Index of socket
+ */
+static int get_socket_idx(struct modem_socket *sock)
+{
+	for (int i = 0; i < mdata.socket_config.sockets_len; i++) {
+		if (&mdata.socket_config.sockets[i] == sock) {
+			return i;
+		}
+	}
+	return -1;
+}
+#endif
+
+/**
+ * @brief Handler for OK
+ */
+MODEM_CMD_DEFINE(on_cmd_ok)
+{
+	modem_cmd_handler_set_error(data, 0);
+	k_sem_give(&mdata.sem_response);
+	return 0;
+}
+
+/**
+ * @brief Handler for ERROR
+ */
+MODEM_CMD_DEFINE(on_cmd_error)
+{
+	modem_cmd_handler_set_error(data, -EIO);
+	k_sem_give(&mdata.sem_response);
+	return 0;
+}
+
+/**
+ * @brief Handler for sock sentdata
+ */
+MODEM_CMD_DEFINE(on_cmd_sock_sentdata)
+{
+	if (argc < 2) {
+		return -EAGAIN;
+	}
+
+	int data_len = (int)strtol(argv[1], NULL, 10);
+	return data_len;
+}
+
+/**
+ * @brief Send data over the given socket
+ */
+static ssize_t send_socket_data(struct modem_socket *sock, const struct sockaddr *dst_addr,
+				const char *buf, const size_t buf_len, k_timeout_t timeout)
+{
+	int ret = -1;
```

Comment:
dont really need to init this since it is used only after other assignment later in the function